### PR TITLE
Make other entries generic in `PackageManifest` of `BallerinaToml`

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/27896

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/test/resources/ballerina-toml/other-entries.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+firstField = "yee"
+
+[package]
+org = "foo"
+name = "winery"
+version = "1.0.0"
+newField = "fee"
+
+[userTable]
+name = "bar"
```
Comment (hevayo) [inline compiler/ballerina-lang/src/test/resources/ballerina-toml/other-entries.toml:10]: Can we test with some numbers and also some complex tables as well

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java [hunk 2]
```diff
@@ -170,13 +171,10 @@ private PackageManifest parseAsPackageManifest() {
...
-        Map<String, TopLevelNode> otherEntries = new HashMap<>();
+        Map<String, Object> otherEntries = new HashMap<>();
         if (!tomlAstNode.entries().isEmpty()) {
-
-            for (Map.Entry<String, TopLevelNode> entry : tomlAstNode.entries().entrySet()) {
-                if (entry.getKey().equals(PACKAGE)) {
-                    continue;
-                }
+            Map<String, Object> tomlMap = ballerinaToml.toml().toMap();
+            for (Map.Entry<String, Object> entry : tomlMap.entrySet()) {
```
Comment (xlight05) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:177]: Does otherEntries contain [Package] table as well? is that the expected behavior?

Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:177]: @hevayo `[package]` table also can have other entries as we discussed earlier. Should we avoid that?

File: compiler/ballerina-lang/src/test/resources/ballerina-toml/other-entries.toml (hunk 1 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/test/resources/ballerina-toml/other-entries.toml:10]: Added

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java (hunk 2 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:177]: `[package]` and `[platform]` are kept as reserved and these fields won't be added to the other entries

Comment (xlight05) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:177]: Great :)

**Comment by xlight05** [comment]
LGTM

**Review by xlight05 — APPROVED** [review]

**Review by hevayo — APPROVED** [review]

**Pull request merged**